src/profile/jwks.ts

```diff
@@ -0,0 +1,185 @@
+ * @returns A modified copy of the profile document, with the JWKS IRI set.
+ */
+export function setProfileJwks<Dataset extends SolidDataset>(
+  profileDocument: Dataset,
+  webId: WebId,
+  jwksIri: Iri | IriString
+): Dataset {
+  return setThing(
+    profileDocument,
+    setIri(
+      getProfileFromProfileDoc(profileDocument, webId),
+      security.publicKey,
+      jwksIri
+    )
+  );
+}
+
+/**
+ * Look for a JWKS IRI optionally advertized from a profile document.
+ *
+ * @param profileDocument The profile document.
+ * @param webId The WebID featured in the profile document.
+ * @returns The JWKS IRI associated to the WebID, if
+ */
+export function getProfileJwksIri(
+  profileDocument: SolidDataset,
+  webId: WebId
+): UrlString | null {
+  return getUrl(
+    getProfileFromProfileDoc(profileDocument, webId),
+    security.publicKey
+  );
+}
+
+const isJwks = (jwksDocument: Jwks | unknown): jwksDocument is Jwks => {
+  return typeof (jwksDocument as Jwks).keys !== "undefined";
+};
+
+/**
+ * Fetch a JWKS at a given IRI, and add the given JWK to the obtained key set.
+ *
+ * @param jwk The JWK to add to the set.
+ * @param jwksIri The IRI where the key set should be looked up.
+ * @param options @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ * @returns Promise resolving to a JWKS where the given key has been added.
+ */
+export async function addJwkToJwks(
+  jwk: Jwk,
+  jwksIri: IriString,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): Promise<Jwks> {
+  const config = {
+    ...internal_defaultFetchOptions,
+    ...options,
+  };
+  const jwksResponse = await config.fetch(jwksIri);
+  if (!jwksResponse.ok) {
+    throw new Error(
+      `Fetching [${jwksIri}] returned an error: ${jwksResponse.status} ${jwksResponse.statusText}`
+    );
+  }
+  try {
+    const jwksDocument = await jwksResponse.json();
+    if (!isJwks(jwksDocument)) {
+      throw new Error(
+        `${jwksIri} did not dereference to a valid JWKS: ${JSON.stringify(
+          jwksDocument
+        )}`
+      );
+    }
+    return {
+      keys: [...jwksDocument.keys, jwk],
+    };
+  } catch (e) {
+    throw new Error(`Parsing the document at [${jwksIri}] failed: ${e}`);
+  }
+}
+
+/**
+ * Adds a public key to the JWKS listed in the profile associated to the given WebID.
+ * After having looked up the profile document thanks to the WebID, the associated
+ * JWKS, and having added the given key to the JWKS, this function overwrites the
+ * previous JWKS so that the new version is saved. This assumes the JWKS is hosted
+ * at a read-write IRI, such as a Solid pod.
+ *
+ * @param publicKey The public key value to set.
+ * @param webId The WebID of the profile to add the public key.
```

Comment:
I think this reads a wee bit funny - i.e., it's not that Profile Documents have WebIDs. Suggest this:
```suggestion
 * @param webId The WebID whose profile document references the key set to which we wish to add the specified public key.
```